File: problem03.py
import math

def is_prime(n):
    if n < 0:
        return False
    for j in range(2, math.sqrt(n)+1):
        if n%j == 0:
            return False
    return True

big = 600851475143

i = 1

while i*i <= big:
    i += 2
    if big % i == 0:
        if is_prime(i):
            maximum = i

print maximum

File: go/problem03.go
package main

import (
    "fmt"
    "math"
)

func is_prime(i int64) bool {
    var j int64 = 0;
    if i < 2 {
        return false
    } else if i == 2 {
        return true
    }
    for j = 3; j < sqrt(i); j+=2 {
        if i % j == 0 {
            return false
        }
    }
    return true
}

func sqrt(i int64) int64 {
    return int64(math.Sqrt(float64(i)))
}

func main() {
    var i, largest, num int64 = 0, 0, 600851475143
    for i = 3; i < sqrt(num); i+= 2 {
        if num % i == 0 && is_prime(i) {
            largest = i
        }
    }
    fmt.Printf("Largest: %v\n", largest)
}
